core/hw/sh4/dyna/ngen.h

```diff
@@ -95,7 +95,8 @@ u32 DYNACALL rdv_DoInterrupts_pc(u32 pc);
 
 //Stuff to be implemented per dynarec core
 
-void ngen_init();
+typedef void (*t_mainloop_fn) (void*);
+t_mainloop_fn ngen_init();
```

Comment:
Ideally we would return an `ngen_interface_t` here. Or passing it as an `ngen_interface_t* if` out parameter